이슈, 코멘트 API 구현 / 일부 외래키 설정 변경

seokju2ng opened the pull request:
### 코멘트 API
- 추가 API 구현: userNum은 인증정보에서 가져오게 수정되어야 함
- 수정 API 구현

### 이슈 API
- 추가 API 구현
  - 이슈등록 > 코멘트등록 > 어싸이니등록 > 레이블등록 순으로 진행
  - userNum은 인증정보에서 가져오게 수정되어야 함
  - Labeling과 Assignees 등록은 서비스가 나오면 붙이는 작업 요구

### 모델 외래키 Not Null
- 유저 - 이슈
- 유저 - 코멘트
- 이슈 - 코멘트

### 일부 외래키명을 카멜케이스로 변경
- 코멘트, 이슈, 마일스톤, 유저

seokju2ng committed:
[server] feat: 코멘트 추가 API

- 코멘트 추가 API 완성
- 유저번호는 인증 정보에서 가져와야함

seokju2ng committed:
[server] feat: comment 모델 외래키 NotNull

- null이 들어올 수 없는 컬럼인 외래키(유저키, 이슈키)에 대해서
컨트롤러에서 처리해주기보단, 디비에 insert쿼리가 실패했을
때 에러메시지를 반환
- 유저와 이슈, 코멘트 모델에 외래키 notnull옵션을 추가

seokju2ng committed:
[server] feat: 코멘트 수정 API 구현

- 업데이트에 성공하면 200 응답
- 업데이트에 실패하면 404 응답
  - 요청한 번호가 없을 때
  - content를 안보냈을 때

seokju2ng committed:
[server] feat: 이슈 추가 API 구현

- 이슈와 유저 관계 외래키를 not null 설정
- 이슈등록 > 코멘트등록 > 어싸이니등록 > 레이블등록 순으로 진행
- 레이블과 어싸이니 등록은 서비스가 나오면 붙이는 작업 요구

seokju2ng committed:
[server] feat: 외래키명을 Camel Case로 변경

- 일부 외래키명을 camel case로 변경

skid901 commented on web/server/src/controllers/issues.js:
```diff
@@ -15,6 +16,19 @@ const issueController = {
+    }
+    const issue = await issueService.add({ title, userNum, milestoneNum });
+    await commentService.add({ content, userNum, issueNum: issue.num });
+    // assignees 피봇 테이블에 등록
+    // labeling 피봇 테이블에 등록
```
```javascript
await labelingService.add({ issueNum: issue.num, labelNum });
await assignmentService.add({ issueNum: issue.num, userNum });
```
추후 위와 같이 작업 진행하면 될 것 같습니다.
고생하셨습니다 ^^7

ahrimy reviewed (approved):
에러처리까지 깔끔하게 잘 구현됐네용 이후에 assignee, label 지정만 추가하면 될 것 같아요!

ahrimy commented:
conflict 난 것부터 해결을 해야겠군요......

seokju2ng committed:
Merge branch 'dev_web' into feat/server/API-issue-comment

seokju2ng committed:
[server] feat: Conflict 해결 및 이슈 추가 API 구현 완료

- 피봇 테이블 labelings와 assignments에 데이터 추가까지 완료
- 충돌난 부분 전부 해결

seokju2ng commented on web/server/src/controllers/issues.js:
```diff
@@ -15,6 +18,26 @@ const issueController = {
+    }
+    const issue = await issueService.add({ title, userNum, milestoneNum });
+    await commentService.add({ content, userNum, issueNum: issue.num });
+    labels.forEach(
+      async labelNum =>
+        await labelingService.add({ issueNum: issue.num, labelNum }),
+    );
+    assignees.forEach(
+      async userNum =>
+        await assignmentService.add({ issueNum: issue.num, userNum }),
+    );
```
라벨링과 어싸인먼트의 추가 기능까지 삽입되었습니다 ^^7

seokju2ng reviewed (commented):
충돌 해결하고 이슈 추가 API 완료했습니다 확인 한번씩 부탁드려욥

seokju2ng committed:
[server] feat: issueController 피봇 등록 await, async 제거

- 없는 키를 등록하려고 하면 다른 에러 없이 성공을 응답하는 문제
- 트랜잭션 적용 전까지는 그냥 없으면 등록안되는 지금 코드로 두기로 결정
